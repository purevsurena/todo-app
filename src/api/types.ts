export interface ApiResponse<T> {
  data: T;
  error?: string;
}

export type PhotoMetadata = {
  title: string;
  tags: string[];
  date: string;
};
