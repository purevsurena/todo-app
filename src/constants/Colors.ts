/**
 * Below are the colors that are used in the app. The colors are defined in the light and dark mode.
 * There are many other ways to style your app. For example, [Nativewind](https://www.nativewind.dev/), [Tamagui](https://tamagui.dev/), [unistyles](https://reactnativeunistyles.vercel.app), etc.
 */

const tintColorLight = "#0a7ea4";
const tintColorDark = "#fff";

export const Colors = {
  light: {
    text: "#11181C",
    background: "#fff",
    tint: tintColorLight,
    icon: "#687076",
    tabIconDefault: "#687076",
    tabIconSelected: tintColorLight,
  },
  dark: {
    text: "#ECEDEE",
    background: "#151718",
    tint: tintColorDark,
    icon: "#9BA1A6",
    tabIconDefault: "#9BA1A6",
    tabIconSelected: tintColorDark,    
  },
  generic: {
    dark: "#000",
    background: '#2c2c2e',
    cardBG: '#1c1c1e',
    greyText: '#9E9EA5',
    blue: '#54a0ff',
    orange: '#ff5722',
    turquoise: '#00adb5',
    yellow: '#FFFF00',
  }
};
