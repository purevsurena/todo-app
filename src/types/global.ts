// types.d.ts
export interface Task {
  _id: string;
  title: string;
  category: string;
  dueDate: string;
  completed: boolean;
  userId: string;
}

export type GategoryTypes =
  | "quick_notes"
  | "today"
  | "tomorrow"
  | "archive"
  | "wishlist";

export type BottomSheetModalLabelProps = {
  id: number;
  label: string;
  category: GategoryTypes;
};
