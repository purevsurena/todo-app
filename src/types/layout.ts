import { ViewStyle } from "react-native";

// WhiteSpace props
export type WhiteSpaceSizeProps =
  | "0"
  | "2"
  | "4"
  | "6"
  | "8"
  | "10"
  | "12"
  | "16"
  | "20"
  | "24"
  | "28"
  | "32"
  | "36"
  | "40"
  | "48"
  | "56"
  | "64"
  | "72"
  | "80"
  | "88"
  | "96";

export type WhiteSpaceProps = {
  size: WhiteSpaceSizeProps;
  style?: ViewStyle;
};
