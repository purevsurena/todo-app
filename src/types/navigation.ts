import { RouteProp } from "@react-navigation/native";

export interface DodoListDetailScreenProps {
  type: "today" | "tomorrow" | "archive" | "wishlist" | "quick_notes" | "all";
}

export type RootStackParamList = {
  Home: undefined;
  "todo-list-detail": {
    type: "today" | "tomorrow" | "archive" | "wishlist" | "quick_notes" | "all";
  };
  // Add other screens here as needed
};

declare global {
  namespace ReactNavigation {
    interface RootParamList extends RootStackParamList {}
  }
}

export type TodoListDetailRouteProp = RouteProp<
  { params: DodoListDetailScreenProps },
  "params"
>;
