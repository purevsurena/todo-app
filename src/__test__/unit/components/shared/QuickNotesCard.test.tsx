import React from "react";
import { render } from "@testing-library/react-native";
import QuickNotesCard from "@/components/shared/QuckNotesCard";
import { Colors } from "@/constants/Colors";
import { Task } from "@/types/global";

describe("QuickNotesCard Component", () => {
  const tasks: Task[] = [
    {
      _id: "1",
      title: "Task 1",
      category: "today",
      dueDate: "2024-07-26T00:00:00.000Z",
      completed: false,
      userId: "user-1",
    },
    {
      _id: "2",
      title: "Task 2",
      category: "tomorrow",
      dueDate: "2024-07-27T00:00:00.000Z",
      completed: true,
      userId: "user-1",
    },
  ];

  it("should render the card with the correct title, subtitle, and tasks", () => {
    const { getByTestId } = render(
      <QuickNotesCard
        iconName="note"
        iconColor={Colors.generic.blue}
        title="Quick Notes"
        subtitle="Today"
        tasks={tasks}
        onPress={() => {}}
      />,
    );

    expect(getByTestId("cardTitle").props.children).toBe("Quick Notes");
    expect(getByTestId("cardSubtitle").props.children).toBe("Today");
    expect(getByTestId("task-0").props.children).toBe("Task 1");
    expect(getByTestId("task-1").props.children).toBe("Task 2");
  });

  it("matches the snapshot", () => {
    const { toJSON } = render(
      <QuickNotesCard
        iconName="note"
        iconColor={Colors.generic.blue}
        title="Quick Notes"
        subtitle="Today"
        tasks={tasks}
        onPress={() => {}}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
