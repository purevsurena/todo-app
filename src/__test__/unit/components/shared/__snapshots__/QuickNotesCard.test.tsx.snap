// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuickNotesCard Component matches the snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "backgroundColor": "#1c1c1e",
        "borderRadius": 16,
        "padding": 12,
        "paddingTop": 24,
        "paddingVertical": 16,
        "width": 718,
      },
    ]
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 12,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#54a0ff",
            "fontSize": 30,
          },
          undefined,
          {
            "fontFamily": "material-community",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
      testID="icon"
    >
      ó°Žš
    </Text>
    <Text
      style={
        [
          {
            "color": "#fff",
          },
          undefined,
          {
            "fontSize": 32,
            "fontWeight": "bold",
            "lineHeight": 32,
          },
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
      testID="cardTitle"
    >
      Quick Notes
    </Text>
  </View>
  <Text
    style={
      [
        {
          "color": "#54a0ff",
        },
        undefined,
        undefined,
        undefined,
        {
          "fontSize": 20,
          "fontWeight": "bold",
        },
        undefined,
        undefined,
      ]
    }
    testID="cardSubtitle"
  >
    Today
  </Text>
  <View
    style={
      [
        {
          "height": 4,
        },
        undefined,
      ]
    }
    testID="white-space"
  />
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "backgroundColor": "#2c2c2e",
          "borderRadius": 12,
          "marginTop": 4,
          "padding": 8,
          "paddingHorizontal": 12,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#54a0ff",
            "borderRadius": 3,
            "height": 6,
            "marginRight": 6,
            "width": 6,
          }
        }
      />
      <Text
        numberOfLines={1}
        style={
          [
            {
              "color": "#fff",
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            {
              "color": "#9E9EA5",
              "flex": 1,
              "fontSize": 13,
              "fontWeight": "500",
            },
          ]
        }
        testID="task-0"
      >
        Task 1
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#54a0ff",
            "borderRadius": 3,
            "height": 6,
            "marginRight": 6,
            "width": 6,
          }
        }
      />
      <Text
        numberOfLines={1}
        style={
          [
            {
              "color": "#fff",
            },
            {
              "fontSize": 16,
              "lineHeight": 24,
            },
            undefined,
            undefined,
            undefined,
            undefined,
            {
              "color": "#9E9EA5",
              "flex": 1,
              "fontSize": 13,
              "fontWeight": "500",
            },
          ]
        }
        testID="task-1"
      >
        Task 2
      </Text>
    </View>
  </View>
</View>
`;
